xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -181,16 +179,36 @@ static int AEChannelMapToAUDIOTRACKChannelMask(CAEChannelInfo info)
 jni::CJNIAudioTrack *CAESinkAUDIOTRACK::CreateAudioTrack(int stream, int sampleRate, int channelMask, int encoding, int bufferSize)
 {
   jni::CJNIAudioTrack *jniAt = NULL;
-  m_jniAudioFormat = encoding;
 
   try
   {
-    jniAt = new CJNIAudioTrack(stream,
-                               sampleRate,
-                               channelMask,
-                               encoding,
-                               bufferSize,
-                               CJNIAudioTrack::MODE_STREAM);
+    if (CJNIBase::GetSDKVersion() >= 21)
+    {
+      CJNIAudioAttributesBuilder attrBuilder;
+      attrBuilder.setContentType(stream == CJNIAudioManager::STREAM_MUSIC
+        ? CJNIAudioAttributes::CONTENT_TYPE_MUSIC
+        : CJNIAudioAttributes::CONTENT_TYPE_MOVIE);
+      attrBuilder.setUsage(CJNIAudioAttributes::USAGE_MEDIA);
+      attrBuilder.setFlags(CJNIAudioAttributes::FLAG_AUDIBILITY_ENFORCED);
+
+      CJNIAudioFormatBuilder fmtBuilder;
+      fmtBuilder.setChannelMask(channelMask);
+      fmtBuilder.setEncoding(encoding);
+      fmtBuilder.setSampleRate(sampleRate);
+
+      jniAt = new CJNIAudioTrack(attrBuilder.build(),
+                                 fmtBuilder.build(),
+                                 bufferSize,
+                                 CJNIAudioTrack::MODE_STREAM,
+                                 CJNIAudioManager::AUDIO_SESSION_ID_GENERATE);
+    }
+    else
+      jniAt = new CJNIAudioTrack(stream,
```

Comment:
since the "if" is within parenthesis, wouldn't it be nicer to also have the "else" condition use them? Especially when you split the "one liner" to multiple lines. But maybe it's just me that finds the current formatting a bit harder to read.

Reply:
You can drop the else entirely. We only support SDKVersion >= 21